### ISSUE TITLE
Set/get the InferenceModel index embeddings, and save/load the index

### ISSUE THREAD

**Issue by mosheliv** [KevinMusgrave/pytorch-metric-learning#304]
is there a way to get the embeddings and save them after calling the inference.train_indexer? I really dont want to retrain the indexer each time the app starts. I looked at the code and could't see a way to set/get the list of embeddings or add new ones to the already trained indexer.

from what I see, the FaissIndexer class needs to be extended with get_index, set_index and add_to_index

**Comment by KevinMusgrave** [comment]
The inference module is a bit limited currently. The features you've mentioned would be good to add in the future though.

As a temporary workaround, I think you can maybe use [```faiss.write_index``` and ```faiss.read_index```](https://github.com/facebookresearch/faiss/wiki/Index-IO,-cloning-and-hyper-parameter-tuning#io-and-deep-copying-indexes) and [```index.add```](https://github.com/facebookresearch/faiss/wiki/Getting-started#building-an-index-and-adding-the-vectors-to-it) like this:

```python
import faiss

# save faiss index
faiss.write_index(inferece.indexer.index, "blah.index")

# load faiss index
inference.indexer.index = faiss.read_index("blah.index")

# add embeddings to index
inference.indexer.index.add(new_embeddings)
```

**Comment by mosheliv** [comment]
Thank you, that is a pretty good workaround.

**Comment by KevinMusgrave** [comment]
Re-opening until this gets implemented.
